### PR TITLE
Updated tor.nim

### PR DESCRIPTION
**Pull request by Techno-Fox**
Added more network functions, made ProxySocket public, and made for versitile.

### CODE REVIEW

File: tor.nim [hunk 1]
```diff
@@ -237,18 +239,117 @@ proc connect*(s: ProxySocket, address: string, port: Port) =
   p[0] = cast[char](port.uint16 shr 8)
   p[1] = cast[char](port)
 
+  echo "Connecting through tor"
   s.inner.send("\x05\x01\x00\x03" & address.len.char & address & p)
 
+proc send*(s: ProxySocket; data: pointer; size: int): int = 
+  #[
```
Comment (FedericoCeratto) [inline tor.nim]: Can you please switch comments to the usual "##" ?

File: tor.nim [hunk 2]
```diff
@@ -237,18 +239,117 @@ proc connect*(s: ProxySocket, address: string, port: Port) =
...
 proc send*(s: ProxySocket; data: string; flags = {SafeDisconn}) =
-  ## Sends data to a socket
+  #[
+    Sends data to a socket
+  ]#
   s.inner.send(data, flags)
 
+proc trySend*(s: ProxySocket; data: string): bool = 
+  #[
+    Safe alternative to send. Does not raise an OSError when an error occurs, and instead returns false on failure.
```
Comment (FedericoCeratto) [inline tor.nim]: Please remove "Safe"

File: tor.nimble [hunk 3]
```diff
@@ -1,10 +1,10 @@
 # Package
 
-version       = "0.1.0"
-author        = "Federico Ceratto"
+version       = "1.1.0"
```
Comment (FedericoCeratto) [inline tor.nimble]: 0.2.0 please :)

File: tor.nimble [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
+author        = "Federico Ceratto and KittyTechno"
 description   = "Tor helper library"
 license       = "MPL-2.0"
 
 # Dependencies
 
-requires "nim >= 0.20.0"
+requires "nim >= 1.2.6"
```
Comment (FedericoCeratto) [inline tor.nimble:10]: Why this version bump? Let's use 1.0.0 unless there are real compatibility issues

**Review by FedericoCeratto — CHANGES_REQUESTED** [review]
Thanks for the PR. Few minor changes and it's good to go.

File: tor.nim (hunk 2 above)
Comment (Techno-Fox) [reply tor.nim]: sorry I literaly copied and pasted from the net module

File: tor.nimble (hunk 4 above)
Comment (Techno-Fox) [reply tor.nimble:10]: I put the version bump, because thats the current version of nim-lang I was developing in.

File: tor.nim (hunk 1 above)
Comment (Techno-Fox) [reply tor.nim]: I can

File: tor.nimble (hunk 4 above)
Comment (Techno-Fox) [reply tor.nimble:10]: Thats the current stable version of nim-lang, and is the current version I'm using.

File: tor.nimble (hunk 3 above)
Comment (Techno-Fox) [reply tor.nimble]: alright

**Comment by Techno-Fox** [comment]
Ok. The only one I haven't changed is the version bump for nim-lang. As that is the current stable version.

**Pull request merged**